pkg/service/backup/purger.go

```diff
@@ -110,9 +126,47 @@ func (p purger) PurgeSnapshotTags(ctx context.Context, manifests []*ManifestInfo
 			}
 		}
 	}
-	if _, err := p.deleteFiles(ctx, manifests[0].Location, files); err != nil {
-		return 0, errors.Wrapf(err, "delete")
+	// Purge versioned SSTables
+	if !oldest.IsZero() {
+		nodeDir := RemoteSSTableBaseDir(anyM.ClusterID, anyM.DC, anyM.NodeID)
+		opts := &scyllaclient.RcloneListDirOpts{
+			FilesOnly:     true,
+			Recurse:       true,
+			VersionedOnly: true,
+		}
+
+		cb := func(item *scyllaclient.RcloneListDirItem) {
+			snapshotSuffix := path.Ext(item.Name)[1:]
+			t, err := SnapshotTagTime(snapshotSuffix)
```

Comment:
I don't like it, because there is a hidden assumption made in the code.
I know that the file extension is the snapshot_tag, but I know it because I'm going through the PR and I saw the part where you defined in `rc.go`
```
const snapshotTagSuffixRegex = `{**.sm_*UTC}`
```
Please wrap the versioning related code into `pkg/util/versioning` package or anything else what will define it's single responsibility.
`pkg/util/versioning` should expose everything related to your versioning implementation including`snapshotTagSuffixRegex` const from `rc.go`

Reply:
I added it to `service/backup/backupspec` because it seems to better fit there.